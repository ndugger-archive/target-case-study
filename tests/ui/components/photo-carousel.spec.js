import React from 'react';
import { shallow } from 'enzyme';

import PhotoCarousel from '../../../src/ui/components/photo-carousel';

const photos = [ 'foo', 'bar', 'baz', 'car', 'caz' ];

test('carousel -> should match snapshot', () => {
    const carousel = shallow(<PhotoCarousel photos={ photos }/>);

    expect(carousel).toMatchSnapshot();
});

test('carousel.thumbnails -> should be an array of [ previous, current, next ]', () => {
    const carousel = shallow(<PhotoCarousel photos={ photos }/>);
    const instance = carousel.instance();

    expect(instance.thumbnails).toEqual([ 'caz', 'foo', 'bar' ]);
});

test('carousel.state.current -> initial value should be 0', () => {
    const carousel = shallow(<PhotoCarousel photos={ photos }/>);
    const instance = carousel.instance();

    expect(instance.state.current).toEqual(0);
})

test('carousel.handleThumbnailClick -> should set the clicked thumbnail as the new current photo', () => {
    const carousel = shallow(<PhotoCarousel photos={ photos }/>);
    const instance = carousel.instance();

    instance.handleThumbnailClick('bar');

    expect(instance.state.current).toEqual(1);
});

test('carousel.handleNavPrevClick -> should navigate the thumbnails back one', () => {
    const carousel = shallow(<PhotoCarousel photos={ photos }/>);
    const instance = carousel.instance();

    instance.handleNavPrevClick();

    expect(instance.state.current).toEqual(4);
});

test('carousel.handleNavNextClick -> should navigate the thumbnails forward one', () => {
    const carousel = shallow(<PhotoCarousel photos={ photos }/>);
    const instance = carousel.instance();

    instance.handleNavNextClick();

    expect(instance.state.current).toEqual(1);
});
