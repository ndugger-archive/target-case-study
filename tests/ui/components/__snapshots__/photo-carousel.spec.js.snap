exports[`test carousel -> should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div
        style={
            Object {
                "backgroundImage": "url(foo)",
              }
        } />
    <styled.div>
        <Styled(Icon)
            glyph="chevron_left"
            onClick={[Function]} />
        <styled.div
            current={false}
            onClick={[Function]}
            style={
                Object {
                    "backgroundImage": "url(caz)",
                  }
            } />
        <styled.div
            current={true}
            onClick={[Function]}
            style={
                Object {
                    "backgroundImage": "url(foo)",
                  }
            } />
        <styled.div
            current={false}
            onClick={[Function]}
            style={
                Object {
                    "backgroundImage": "url(bar)",
                  }
            } />
        <Styled(Icon)
            glyph="chevron_right"
            onClick={[Function]} />
    </styled.div>
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div
            style={
                  Object {
                        "backgroundImage": "url(foo)",
                      }
            } />
      <styled.div>
            <Styled(Icon)
                  glyph="chevron_left"
                  onClick={[Function]} />
            <styled.div
                  current={false}
                  onClick={[Function]}
                  style={
                        Object {
                              "backgroundImage": "url(caz)",
                            }
                  } />
            <styled.div
                  current={true}
                  onClick={[Function]}
                  style={
                        Object {
                              "backgroundImage": "url(foo)",
                            }
                  } />
            <styled.div
                  current={false}
                  onClick={[Function]}
                  style={
                        Object {
                              "backgroundImage": "url(bar)",
                            }
                  } />
            <Styled(Icon)
                  glyph="chevron_right"
                  onClick={[Function]} />
      </styled.div>
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <PhotoCarousel
        photos={
                Array [
                        "foo",
                        "bar",
                        "baz",
                        "car",
                        "caz",
                      ]
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PhotoCarousel {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "photos": Array [
            "foo",
            "bar",
            "baz",
            "car",
            "caz",
          ],
        },
        "refs": Object {},
        "state": Object {
          "current": 0,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div
                    style={
                              Object {
                                        "backgroundImage": "url(foo)",
                                      }
                    } />
          <styled.div>
                    <Styled(Icon)
                              glyph="chevron_left"
                              onClick={[Function]} />
                    <styled.div
                              current={false}
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "backgroundImage": "url(caz)",
                                                }
                              } />
                    <styled.div
                              current={true}
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "backgroundImage": "url(foo)",
                                                }
                              } />
                    <styled.div
                              current={false}
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "backgroundImage": "url(bar)",
                                                }
                              } />
                    <Styled(Icon)
                              glyph="chevron_right"
                              onClick={[Function]} />
          </styled.div>
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div
                    style={
                              Object {
                                        "backgroundImage": "url(foo)",
                                      }
                    } />
          <styled.div>
                    <Styled(Icon)
                              glyph="chevron_left"
                              onClick={[Function]} />
                    <styled.div
                              current={false}
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "backgroundImage": "url(caz)",
                                                }
                              } />
                    <styled.div
                              current={true}
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "backgroundImage": "url(foo)",
                                                }
                              } />
                    <styled.div
                              current={false}
                              onClick={[Function]}
                              style={
                                        Object {
                                                  "backgroundImage": "url(bar)",
                                                }
                              } />
                    <Styled(Icon)
                              glyph="chevron_right"
                              onClick={[Function]} />
          </styled.div>
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PhotoCarousel
    photos={
        Array [
            "foo",
            "bar",
            "baz",
            "car",
            "caz",
          ]
    } />,
}
`;
